crates/functional-yew/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+        o3,
+        o4,
+        (),
+    );
+}
+
+pub fn use_effect5<F, Destructor, T1, T2, T3, T4, T5>(
+    callback: Box<F>,
+    o1: T1,
+    o2: T2,
+    o3: T3,
+    o4: T4,
+    o5: T5,
+) where
+    F: FnOnce(&T1, &T2, &T3, &T4, &T5) -> Destructor,
+    Destructor: FnOnce() + 'static,
+    T1: PartialEq + 'static,
+    T2: PartialEq + 'static,
+    T3: PartialEq + 'static,
+    T4: PartialEq + 'static,
+    T5: PartialEq + 'static,
+{
+    struct UseEffectState<T1, T2, T3, T4, T5, Destructor> {
+        o1: Rc<T1>,
+        o2: Rc<T2>,
+        o3: Rc<T3>,
+        o4: Rc<T4>,
+        o5: Rc<T5>,
+        destructor: Option<Box<Destructor>>,
+    }
+    let o1 = Rc::new(o1);
+    let o2 = Rc::new(o2);
+    let o3 = Rc::new(o3);
+    let o4 = Rc::new(o4);
+    let o5 = Rc::new(o5);
+    let o1_c = o1.clone();
+    let o2_c = o2.clone();
+    let o3_c = o3.clone();
+    let o4_c = o4.clone();
+    let o5_c = o5.clone();
+    use_hook(
+        move |state: &mut UseEffectState<T1, T2, T3, T4, T5, Destructor>, hook_update| {
+            let mut should_update = !(*state.o1 == *o1
+                && *state.o2 == *o2
+                && *state.o3 == *o3
+                && *state.o4 == *o4
+                && *state.o5 == *o5);
+
+            if should_update {
+                if let Some(de) = state.destructor.take() {
+                    de();
+                }
+                let new_destructor = callback(
+                    o1.borrow(),
+                    o2.borrow(),
+                    o3.borrow(),
+                    o4.borrow(),
+                    o5.borrow(),
+                );
+                state.o1 = o1.clone();
+                state.o2 = o2.clone();
+                state.o3 = o3.clone();
+                state.o4 = o4.clone();
+                state.o5 = o5.clone();
+                state.destructor.replace(Box::new(new_destructor));
+            } else if state.destructor.is_none() {
+                should_update = true;
+                state.destructor.replace(Box::new(callback(
+                    state.o1.borrow(),
+                    state.o2.borrow(),
+                    state.o3.borrow(),
+                    state.o4.borrow(),
+                    state.o5.borrow(),
+                )));
+            }
+            return move || {
+                if should_update {
+                    hook_update(move |_: &mut UseEffectState<T1, T2, T3, T4, T5, Destructor>| true)
+                }
+            };
+        },
+        || UseEffectState {
+            o1: o1_c,
+            o2: o2_c,
+            o3: o3_c,
+            o4: o4_c,
+            o5: o5_c,
+            destructor: None,
+        },
+    )();
+}
+
+pub fn use_hook<InternalHookState, HookRunner, R, InitialStateProvider, PretriggerChange: 'static>(
+    hook_runner: HookRunner,
+    initial_state_producer: InitialStateProvider,
+) -> R
+where
+    HookRunner: FnOnce(&mut InternalHookState, Rc<dyn Fn(PretriggerChange)>) -> R,
```

Comment:
nit: this `Rc` can be a `Box`

Reply:
It used to be an Rc because of something before the overhaul I think. Fixed!